mesa/visualization/modules/CanvasGridVisualization.py

```diff
@@ -25,9 +25,13 @@ class CanvasGrid(VisualizationElement):
             For Circles:
                 "r": The radius, defined as a fraction of cell size. r=1 will
                      fill the entire cell.
+                "x_align", "y_align": Alignment of the circle within the cell.
+                                      Defaults to 0.5 (center).
             For Rectangles:
                 "w", "h": The width and height of the rectangle, which are in
                           fractions of cell width and height.
+                "x_align", "y_align": Alignment of the rectangle within the
```

Comment:
Need to be camel case -- `xAlign`, `yAlign`. I know that some of the JS code use underscores, but we should make it consistent for newly added code.

Reply:
Done. Would it also be a good idea to implement the constraints I mentioned to prevent the shapes from going outside the grid cells?

Reply:
You meant, the clamp shouldn't between 0.0 and 1.0, but instead `r` and `1.0 - r`? Yeah sounds good.

Reply:
Done. Tested with both circles and rectangles and it is working as intended.